sms_wsj/kaldi/utils.py

```diff
@@ -146,33 +147,35 @@ def create_data_dir_from_audio_dir(
     """
     if isinstance(id_to_file_name, str):
         if '{}' in id_to_file_name:
+            if isinstance(target_speaker, (list, tuple)) and len(target_speaker) > 1:
+                assert id_to_file_name.count('{}') == 2, id_to_file_name
             id_to_file_name_fn = lambda _id, spk: id_to_file_name.format(_id, spk)
         else:
             id_to_file_name_fn = lambda _id, spk: id_to_file_name.format(
                 id=_id, spk=spk)
     else:
         id_to_file_name_fn = id_to_file_name
     assert callable(id_to_file_name_fn), id_to_file_name_fn
```

Comment:
How about
```python
if isinstance(target_speaker, (list, tuple)) and len(target_speaker) > 1:
    assert id_to_file_name_fn('id1', 'spk1') != id_to_file_name_fn('id1', 'spk2'), (id_to_file_name_fn('id1', 'spk1'), id_to_file_name_fn('id1', 'spk2'))
    assert id_to_file_name_fn('id1', 'spk1') != id_to_file_name_fn('id2', 'spk1'), (id_to_file_name_fn('id1', 'spk1'), id_to_file_name_fn('id2', 'spk1'))
```
instead of you test?

Reply:
you can always check for more things, one could also check whether the speaker_ids are unique and than you can add the same test to the other branch of the if statement and so on, but I do not think those tests are necessary